### PR TITLE
Implement usage of jsondiffpatch library

### PR DESCRIPTION
**Pull request by juliandsg0222**
## Description

This PR solves the issue #254. The changes made were:

- Implemented the `jsondiffpatch` library.
- Implemented the `undoStack` and `redoStack` stacks, to store the done and undone changes.
- Implemented the `basicData` attribute to store a copy of the editor state.
- Updated the `initialize` method for initializing the `basicData` attribute.
- Updated the `clear` method to clean the content of the `undoStack` and `redoStack` stacks.
- Updated the `editorDidUpdate` method in terms of `basicData`.
- Updated the `save` method to make use of the `diff` method from `jsondiffpatch` library and get the difference (_delta_) between the current `basicData` value and the entry changes, in order to add that `delta` in the `undoStack` and update the `basicData` with the entry changes.
-  Updated the `undo` method to make usage of the `jsonpatchFormatter` format from `jsondiffpatch` library, in order to get clear info about the `delta` stored in the `top` of the `undoStack`, and save it in the `redoStack`, in addition, make usage of the `unpatch` method from `jsondiffpatch` library to build the previous state of `basicData`, removing the changes specified in `lastState`(_delta_). In this point, we are deleting the element from the `undoStack` and saving it into `redoStack`.
- Updated the `redo` method to make usage of the `patch` method from `jsondiffpatch` library to build the next state of `basicData` applying the changes contained in `lastRedoState`(_delta_).  In this point, we are deleting the element from the `redoStack`.
- Replaced in the other methods `stack` with `undoStack`.
- Formated the main file.
- Updated the bundle file.
- Ignore tests temporarily.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
  * @typedef {Object} Undo
  * @description Feature's initialization class.
  * @property {Object} editor — Editor.js instance object.
- * @property {Number} maxLength - Max amount of changes recorded by the history stack.
+ * @property {Number} maxLength - Max amount of changes recorded by the history undoStack.
  * @property {Function} onUpdate - Callback called when the user performs an undo or redo action.
- * @property {Boolean} shouldSaveHistory - Defines if the plugin should save the change in the stack
+ * @property {Boolean} shouldSaveHistory - Defines if the plugin should save the change in the undoStack
  * @property {Object} initialItem - Initial data object.
+ * @property {Object} basicData - Saved data object.
```
Comment (L-Zuluaga) [inline src/index.js:17]: Actually, I preferred the `baseData` name, is there a reason why you decided to change it?

File: src/index.js [hunk 2]
```diff
@@ -255,56 +257,31 @@ export default class Undo {
...
-        this.caret.setToBlock(index, 'end');
-      } else if (this.contentChangedInNoFocusBlock(index, nextIndex)) {
-        const { id } = this.blocks.getBlockByIndex(nextIndex);
-
-        await this.blocks.update(id, state[nextIndex].data);
-        this.setCaretIndex(index, caretIndex);
-      }
+      // To build the previous state of 'basicData', removing the changes
+      // specified in 'lastState'
+      unpatch(this.basicData, lastState);
```
Comment (L-Zuluaga) [inline src/index.js:282]: here the **unpatch** function changes `this.basicData`  by reference right? I mean, **lastState** inverse operation is directly applied to `this.basicData` correct?

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]
Left some comments

File: src/index.js (hunk 2 above)
Comment (juliandsg0222) [reply src/index.js:282]: Yes, the **patch** and **unpatch** operations act directly on the `this.basicData`, so we don't need to do an assignment

File: src/index.js (hunk 1 above)
Comment (juliandsg0222) [reply src/index.js:17]: Yes, we need to change that name, maybe the `basicData` confuses because it sounds like it was only a part of the expected data

**Review by L-Zuluaga — APPROVED** [review]

**Pull request merged**